testsuites/syncgateway/functional/tests/test_xattrs.py

```diff
@@ -153,6 +155,142 @@ def test_olddoc_nil(params_from_base_test_setup, sg_conf_name):
 @pytest.mark.sanity
 @pytest.mark.syncgateway
 @pytest.mark.xattrs
+@pytest.mark.changes
+@pytest.mark.session
+@pytest.mark.parametrize('sg_conf_name, number_users, number_docs_per_user, number_of_updates_per_user', [
+    ('xattrs/no_import', 1, 1, 10),
+    ('xattrs/no_import', 100, 10, 10),
+    ('xattrs/no_import', 10, 1000, 10),
+    ('xattrs/no_import', 100, 1000, 2)
+])
+def test_on_demand_doc_processing(params_from_base_test_setup, sg_conf_name, number_users, number_docs_per_user, number_of_updates_per_user):
+    """
+    1. Start Sync Gateway with autoimport disabled, this will force on-demand processing
+    1. Create 100 users (user_0, user_1, ...) on Sync Gateway each with their own channel (user_0_chan, user_1_chan, ...)
+    1. Load 'number_doc_per_user' with channels specified to each user from SDK
+    1. Make sure the docs are imported via GET /db/doc and POST _bulk_get
+    """
+
+    cluster_conf = params_from_base_test_setup['cluster_config']
+    cluster_topology = params_from_base_test_setup['cluster_topology']
+    mode = params_from_base_test_setup['mode']
+    xattrs_enabled = params_from_base_test_setup['xattrs_enabled']
+
+    cbs_url = cluster_topology['couchbase_servers'][0]
+    sg_admin_url = cluster_topology['sync_gateways'][0]['admin']
+    sg_url = cluster_topology['sync_gateways'][0]['public']
+
+    bucket_name = 'data-bucket'
+    sg_db = 'db'
+    cbs_host = host_for_url(cbs_url)
+
+    # This test should only run when using xattr meta storage
+    if not xattrs_enabled:
+        pytest.skip('XATTR tests require --xattrs flag')
+
+    # This test should only run when mode is CC
+    if mode == "DI":
```

Comment:
This probably needs to be lowercase? Test it out with lowercase in the pytest --mode parameter

Reply:
yeap, modified to lower case to be in consitent with other tests